### PR TITLE
feat: add app-owned YouTube subtitle flow with absPlayer-style parsing

### PR DESCRIPTION
**Pull request by ksyasuda**
## Summary
- adds the app-owned YouTube subtitle flow: mpv boots paused, the overlay opens a track picker, selected subtitle files are downloaded, then playback resumes with injected external tracks
- implements absPlayer-style parsing for YouTube timedtext payloads so downloaded tracks become parsed VTT cues that work with the subtitle sidebar, tokenization, and mining flows
- adds yt-dlp metadata probing plus follow-on runtime/library plumbing so YouTube sessions keep canonical title/channel metadata and cover art
- follow-up hardening: surfaces CLI YouTube flow failures through OSD/logging, suppresses false startup subtitle-failure notifications while app-owned acquisition is still running, keeps no-track picker controls disabled on failed continue attempts, and removes duplicate overlay Anki initialization

## Testing
- bun run changelog:lint
- bun test launcher/commands/playback-command.test.ts src/core/services/cli-command.test.ts src/renderer/modals/youtube-track-picker.test.ts src/main/runtime/youtube-primary-subtitle-notification.test.ts
- bun run typecheck

## Checklist
- [x] Added a changelog fragment in `changes/`, or this PR is labeled `skip-changelog`

## Summary by CodeRabbit

* **New Features**
  * YouTube track picker modal and end-to-end YouTube playback flow (probe, acquire, inject, optional retime)
  * Remote-media Anki support (generate audio/images from remote sources)

* **Improvements**
  * README redesigned with “How It Works”, Features, demo badge and clearer Quick Start
  * Centralized app/launcher log management with rotation/pruning
  * Stats UI: channel-based grouping, improved artwork and media detail
  * Better keyboard, mouse and modal interactions (focus/visibility handling)

* **Settings**
  * YouTube playback options (download/generate), picker shortcut (Ctrl+Alt+C), configurable timeouts

### CODE REVIEW

**Comment by ksyasuda** [comment]
Pushed `d65575c` with the CodeRabbit follow-up fixes.

Addressed:
- launcher stop/child-process handling
- YouTube picker cleanup, re-ack, and rejected IPC handling
- YouTube flow pending-state + picker-open cleanup
- probe/download/timedtext edge cases
- startup Yomitan ordering, type cleanup, and deterministic test/docs sync

Verification run:
- targeted Bun test suite for touched files
- `verify_subminer_change.sh --lane docs --lane launcher-plugin --lane runtime-compat --lane core`
- artifacts: `.tmp/skill-verification/subminer-verify-20260322-191541-mukz16`

File: src/main/runtime/youtube-flow.ts [hunk 1]
```diff
@@ -0,0 +1,583 @@
...
+        return;
+      }
+      const osdProgress = createYoutubeFlowOsdProgress(deps.showMpvOsd);
+      osdProgress.setMessage('Downloading subtitles...');
+      try {
+        const primaryTrack = getTrackById(probe.tracks, request.primaryTrackId);
+        if (!primaryTrack) {
+          deps.warn('No primary YouTube subtitle track selected; continuing without subtitles.');
+          return;
+        }
```
[the remark this answers is not in the document]
Comment (ksyasuda) [reply src/main/runtime/youtube-flow.ts]: Checked this against the current control flow and left it unchanged.

That branch sits inside the download-path `try`, and the enclosing `finally` already runs:
- `osdProgress.stop()`
- `releasePlaybackGate(deps)`
- `restoreOverlayInputFocus(deps)`

I verified the suggested inline cleanup by applying it locally. It double-fired the release/resume path, and the runtime test showed duplicated `script-message subminer-autoplay-ready` / `set_property pause no` commands. So this thread is a false positive for the current code shape.

File: src/renderer/modals/youtube-track-picker.ts [hunk 2]
```diff
@@ -0,0 +1,253 @@
...
+    } catch (error) {
+      setStatus(error instanceof Error ? error.message : String(error), true);
+      return;
+    }
+    if (!response.ok) {
+      setStatus(response.message, true);
+      return;
+    }
+    closeYoutubePickerModal();
+  }
```
[the remark this answers is not in the document]
Comment (ksyasuda) [reply src/renderer/modals/youtube-track-picker.ts:171]: Fixed in 207151db. `resolveSelection()` now short-circuits while a request is in flight, disables the picker controls/buttons during the IPC call, and the modal test suite now verifies duplicate clicks only send one resolve request. I also replaced the empty-state `innerHTML` fallback with explicit DOM construction while touching this path.

File: stats/src/components/library/MediaDetailView.test.tsx [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+      totalTokensSeen: 0,
+      totalLinesSeen: 0,
+      totalLookupCount: 0,
+      totalLookupHits: 0,
+      totalYomitanLookupCount: 0,
+      channelName: null,
+    }),
+    'View Anime',
+  );
+});
```
[the remark this answers is not in the document]
Comment (ksyasuda) [reply stats/src/components/library/MediaDetailView.test.tsx:43]: Fixed in 207151db. Added `videoId` to the non-YouTube `MediaDetailView` test fixture as well so both cases stay type-correct.

File: src/main/runtime/anilist-post-watch.ts [hunk 4]
```diff
@@ -130,6 +132,9 @@ export function createMaybeRunAnilistPostWatchUpdateHandler(deps: {
     if (!mediaKey || !deps.hasMpvClient()) {
       return;
     }
+    if (isYoutubeMediaPath(mediaKey)) {
+      return;
+    }
```
[the remark this answers is not in the document]
Comment (ksyasuda) [reply src/main/runtime/anilist-post-watch.ts:137]: Fixed in 207151db. `isYoutubeMediaPath()` now matches only exact YouTube hosts or subdomains via a shared hostname helper, and `youtube-playback.test.ts` covers `https://notyoutube.com/...` as a regression case.

File: stats/src/components/library/MediaDetailView.test.tsx [hunk 5]
```diff
@@ -0,0 +1,41 @@
...
+      totalTokensSeen: 0,
+      totalLinesSeen: 0,
+      totalLookupCount: 0,
+      totalLookupHits: 0,
+      totalYomitanLookupCount: 0,
+      channelName: 'Creator',
+    }),
+    'View Channel',
+  );
+});
```
[the remark this answers is not in the document]
Comment (ksyasuda) [reply stats/src/components/library/MediaDetailView.test.tsx:23]: Fixed in 207151db. Added the missing `videoId` field to the YouTube-backed `MediaDetailView` test fixture so it matches `MediaDetailData["detail"]` during typecheck.

File: src/main.ts [hunk 6]
```diff
@@ -1040,6 +1229,45 @@ const currentMediaTokenizationGate = createCurrentMediaTokenizationGate();
 const startupOsdSequencer = createStartupOsdSequencer({
   showOsd: (message) => showMpvOsd(message),
 });
+const youtubePrimarySubtitleNotificationRuntime = createYoutubePrimarySubtitleNotificationRuntime({
+  getPrimarySubtitleLanguages: () => getResolvedConfig().youtubeSubgen.primarySubLanguages,
+  notifyFailure: (message) => reportYoutubeSubtitleFailure(message),
+  schedule: (fn, delayMs) => setTimeout(fn, delayMs),
+  clearSchedule: clearYoutubePrimarySubtitleNotificationTimer,
+});
```
[the remark this answers is not in the document]
Comment (ksyasuda) [reply src/main.ts:1241]: Fixed in cdb12827. The generic primary-subtitle notification timer is now suppressed while the app-owned YouTube flow is still probing/downloading and restarted only after the flow settles via `setAppOwnedFlowInFlight(...)` in `src/main/runtime/youtube-primary-subtitle-notification.ts:171`, wired around `runYoutubePlaybackFlowMain(...)` in `src/main.ts:949` and `src/main.ts:979`.

File: src/renderer/modals/youtube-track-picker.ts [hunk 7]
```diff
@@ -0,0 +1,280 @@
...
+        setStatus(response.message, true);
+        return;
+      }
+      closeYoutubePickerModal();
+    } catch (error) {
+      setStatus(error instanceof Error ? error.message : String(error), true);
+    } finally {
+      resolveSelectionInFlight = false;
+      setResolveControlsDisabled(false);
+    }
```
[the remark this answers is not in the document]
Comment (ksyasuda) [reply src/renderer/modals/youtube-track-picker.ts:170]: Fixed in cdb12827. The picker now only keeps controls disabled when the modal is still open and the payload has no tracks in `src/renderer/modals/youtube-track-picker.ts:160`, with regression coverage for the failed no-track continue path in `src/renderer/modals/youtube-track-picker.test.ts:487`.

File: src/core/services/cli-command.ts [hunk 8]
```diff
@@ -396,6 +403,12 @@ export function handleCliCommand(
   } else if (args.jellyfin) {
     deps.openJellyfinSetup();
     deps.log('Opened Jellyfin setup flow.');
+  } else if (args.youtubePlay) {
+    void deps.runYoutubePlaybackFlow({
+      url: args.youtubePlay,
+      mode: args.youtubeMode ?? 'download',
+      source,
+    });
```
[the remark this answers is not in the document]
Comment (ksyasuda) [reply src/core/services/cli-command.ts]: Fixed in cdb12827. The `args.youtubePlay` branch now uses `runAsyncWithOsd(...)` so flow failures hit the existing log/OSD path in `src/core/services/cli-command.ts:406`, with regression coverage in `src/core/services/cli-command.test.ts:249`.

File: src/main.ts [hunk 9]
```diff
@@ -3690,6 +3965,21 @@ function destroyTray(): void {
...
+    getAnkiIntegration: () => appState.ankiIntegration,
+    setAnkiIntegration: (integration) => {
+      appState.ankiIntegration = integration as AnkiIntegration | null;
+    },
+    showDesktopNotification,
+    createFieldGroupingCallback: () => createFieldGroupingCallback(),
+    getKnownWordCacheStatePath: () => path.join(USER_DATA_PATH, 'known-words-cache.json'),
+    shouldStartAnkiIntegration: () =>
+      !(appState.initialArgs && isHeadlessInitialCommand(appState.initialArgs)),
+  });
```
[the remark this answers is not in the document]
Comment (ksyasuda) [reply src/main.ts]: Fixed in cdb12827. Removed the duplicate `initializeOverlayAnkiIntegrationCore(...)` call so `initializeOverlayRuntime()` now relies on the bootstrap path as the single initialization path in `src/main.ts:3971`.

File: launcher/commands/playback-command.test.ts [hunk 10]
```diff
@@ -0,0 +1,127 @@
...
+    },
+    scriptPath: '/tmp/subminer',
+    scriptName: 'subminer',
+    mpvSocketPath: '/tmp/subminer.sock',
+    pluginRuntimeConfig: {
+      socketPath: '/tmp/subminer.sock',
+      autoStart: true,
+      autoStartVisibleOverlay: true,
+      autoStartPauseUntilReady: true,
+    },
```
[the remark this answers is not in the document]
Comment (ksyasuda) [reply launcher/commands/playback-command.test.ts:70]: Fixed in cdb12827. The test now disables the generic overlay auto-pause flags so `startPaused` is asserted through the app-owned YouTube path itself in `launcher/commands/playback-command.test.ts:88`.

File: src/main.ts [hunk 11]
```diff
@@ -787,6 +805,182 @@ const appState = createAppState({
...
+
+  try {
+    await youtubeFlowRuntime.runYoutubePlaybackFlow({
+      url: request.url,
+      mode: request.mode,
+    });
+    logger.info(`YouTube playback flow completed from ${request.source}.`);
+  } finally {
+    youtubePrimarySubtitleNotificationRuntime.setAppOwnedFlowInFlight(false);
+  }
```
[the remark this answers is not in the document]
Comment (ksyasuda) [reply src/main.ts:980]: Fixed in 5f6f93cd. The app-owned in-flight flag now wraps the Windows mpv bootstrap and `connect()` steps too, so synchronous failures cannot leave it stuck true in `src/main.ts:949`.

File: src/renderer/modals/youtube-track-picker.ts [hunk 12]
```diff
@@ -0,0 +1,282 @@
...
+    } catch (error) {
+      setStatus(error instanceof Error ? error.message : String(error), true);
+    } finally {
+      resolveSelectionInFlight = false;
+      const shouldKeepDisabled =
+        ctx.state.youtubePickerModalOpen && !(ctx.state.youtubePickerPayload?.hasTracks ?? false);
+      setResolveControlsDisabled(shouldKeepDisabled);
+    }
+  }
+
```
[the remark this answers is not in the document]
Comment (ksyasuda) [reply src/renderer/modals/youtube-track-picker.ts:172]: Fixed in 5f6f93cd. The picker now uses `payloadHasTracks(...)` as the single source of truth for subtitle availability across render/submit/cleanup paths in `src/renderer/modals/youtube-track-picker.ts:12`, and I swapped the clear operations to `replaceChildren()` while touching that flow. Regression coverage is in `src/renderer/modals/youtube-track-picker.test.ts`.

File: src/main/runtime/youtube-primary-subtitle-notification.ts [hunk 13]
```diff
@@ -0,0 +1,180 @@
...
+
+  const activeTrack =
+    trackList.map(normalizeTrack).find((track) => track?.type === 'sub' && track.id === sid) ?? null;
+  if (!activeTrack) {
+    return false;
+  }
+  if (activeTrack.external) {
+    return true;
+  }
+  return matchesPreferredLanguage(activeTrack.lang, preferredLanguages);
```
[the remark this answers is not in the document]
Comment (ksyasuda) [reply src/main/runtime/youtube-primary-subtitle-notification.ts:94]: Fixed in 5f6f93cd. The notifier now honors `track.selected` as a fallback active subtitle signal when `sid` has not arrived yet via `selected` parsing in `src/main/runtime/youtube-primary-subtitle-notification.ts:6` and the updated active-track lookup in `src/main/runtime/youtube-primary-subtitle-notification.ts:74`. Regression coverage is in `src/main/runtime/youtube-primary-subtitle-notification.test.ts`.

File: src/renderer/modals/youtube-track-picker.test.ts [hunk 14]
```diff
@@ -0,0 +1,874 @@
...
+    assert.equal(overlayFocusCalls.length > 0, true);
+    assert.equal(windowFocusCalls.length > 0, true);
+    assert.deepEqual(ignoreCalls, [{ ignore: true, forward: true }]);
+  } finally {
+    Object.defineProperty(globalThis, 'window', { configurable: true, value: originalWindow });
+    Object.defineProperty(globalThis, 'document', { configurable: true, value: originalDocument });
+    Object.defineProperty(globalThis, 'CustomEvent', {
+      configurable: true,
+      value: originalCustomEvent,
+    });
```
[the remark this answers is not in the document]
Comment (ksyasuda) [reply src/renderer/modals/youtube-track-picker.test.ts]: Fixed in 039e2f56. `src/renderer/modals/youtube-track-picker.test.ts` now uses shared `createYoutubePickerDomFixture()` / `setupYoutubePickerTestEnv()` helpers, and `restoreGlobalProp()` deletes injected globals when they were originally absent instead of restoring `undefined` values. Added a regression at src/renderer/modals/youtube-track-picker.test.ts:155 to prove `window`/`document` get removed after restore, with the shared cleanup used by the picker tests starting at src/renderer/modals/youtube-track-picker.test.ts:56.
